src/proto/error.rs

```diff
@@ -14,6 +14,8 @@ pub enum ProtoError {
     SshKey(#[from] ssh_key::Error),
     #[error("Command not supported ({command})")]
     UnsupportedCommand { command: u8 },
+    #[error("Unexpected response received")]
+    UnexpectedResponse,
```

Comment:
It might be worth explicitly handling [`SSH_AGENT_FAILURE`](https://www.ietf.org/archive/id/draft-miller-ssh-agent-14.html#section-3.1) and [`SSH_AGENT_EXTENSION_FAILURE`](https://www.ietf.org/archive/id/draft-miller-ssh-agent-14.html#section-3.8) here, and returning `UnexpectedResponse` only in cases where the server's response doesn't match any of the expected response types.

Reply:
#45 pulls this up to `AgentError` as a `Failure` or `ExtensionFailure`, so I think we can leave ProtoError like this.